### PR TITLE
send email to user when task is assigned to him

### PR DESCRIPTION
**Pull request by Intrast**
**STATUS**
READY

**DESCRIPTION**

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [ ] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
image

### CODE REVIEW

File: app/controllers/account/tasks_controller.rb [hunk 1]
```diff
@@ -78,15 +78,14 @@ def assign
     @project = parent
     @task = @project.tasks.find(params[:id])
     @result = @task.update(assignee_id: assignee_params[:assignee])
-
     respond_to :js
+    TasksMailer.task_assign_to_user_email(@task).deliver
```
Comment (skcc321) [inline app/controllers/account/tasks_controller.rb]: it should fire when @result is updated and has no validation errors

File: spec/mailers/previews/tasks_mailer_preview.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Preview all emails at http://localhost:3000/rails/mailers/tasks_mailer
+class TasksMailerPreview < ActionMailer::Preview
+  def task_assign_to_user_email
+    task = Task.last
```
Comment (skcc321) [inline spec/mailers/previews/tasks_mailer_preview.rb]: what is that?

File: spec/mailers/tasks_mailer_spec.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 require "rails_helper"
 
 RSpec.describe TasksMailer, type: :mailer do
+<<<<<<< HEAD
```
Comment (skcc321) [inline spec/mailers/tasks_mailer_spec.rb]: ???

**Review by skcc321 — CHANGES_REQUESTED** [review]

File: app/controllers/account/tasks_controller.rb (hunk 1 above)
Comment (Intrast) [reply app/controllers/account/tasks_controller.rb]: я шось не дуже розумію

File: app/views/mailers/tasks/task_assign_to_user_email.html.slim [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  .div-row
+    h1 Task Details:
+  - if @task.description.present?
+    .div-row DESCRIPTION:
+    .div-row
+      .p #{@task.description}
+  .div-row
+    .div-cell ASSIGNEE:
+    .div-cell
+      .p #{@task.assignee.full_name}
```
Comment (skcc321) [inline app/views/mailers/tasks/task_assign_to_user_email.html.slim:19]: user @user instead

File: app/controllers/account/tasks_controller.rb [hunk 5]
```diff
@@ -82,15 +82,14 @@ def assign
     @project = parent
     @task = @project.tasks.find(params[:id])
```
Comment (skcc321) [inline app/controllers/account/tasks_controller.rb]: why do you need so many instance vars?

**Review by skcc321 — CHANGES_REQUESTED** [review]

File: app/views/mailers/tasks/task_assign_to_user_email.html.slim [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  .div-row
+    .div-cell PROJECTS:
```
Comment (igorkasyanchuk) [inline app/views/mailers/tasks/task_assign_to_user_email.html.slim]: Projects? maybe without "S"

File: app/mailers/tasks_mailer.rb [hunk 7]
```diff
@@ -1,6 +1,12 @@
 class TasksMailer < ApplicationMailer
   default template_path: 'mailers/tasks'
 
+  def task_assign_to_user_email(task)
+    @task = task
+    @user = @task.assignee
+    mail(to: @user.email, subject: "Task detail")
```
Comment (igorkasyanchuk) [inline app/mailers/tasks_mailer.rb]: change subject to "New Assignment"

File: app/views/mailers/tasks/task_assign_to_user_email.html.slim [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    .title
+      = link_to @task.title, account_project_task_url(@task.project, @task)
+  .div-row
+    .p was assigned to you
+  .div-row
+    h1 Task Details:
+  - if @task.description.present?
+    .div-row DESCRIPTION:
+    .div-row
+      .p #{@task.description}
```
Comment (igorkasyanchuk) [inline app/views/mailers/tasks/task_assign_to_user_email.html.slim]: use simple_format and remove "p"

Comment (Intrast) [reply app/views/mailers/tasks/task_assign_to_user_email.html.slim]: якщо робити через simple_format тоді робляться великі відступи між  рядками

Comment (igorkasyanchuk) [reply app/views/mailers/tasks/task_assign_to_user_email.html.slim]: напиши дескріпшен в 3 рядки окремі
і глянь як вони прийдуть. вони прийдуть як одна стрічка
зроби щоб теж було 3 рядки

File: app/assets/stylesheets/mailer.scss [hunk 9]
```diff
@@ -27,6 +27,10 @@
   }
 }
 
+p, h1 {
+  margin: 0px;
```
Comment (pereslop) [inline app/assets/stylesheets/mailer.scss]: you don't need use 'px' for zero values
I think you your css libraries already have classes for margin: 0 and padding: 0, you can check it and just add them to template

Comment (igorkasyanchuk) [reply app/assets/stylesheets/mailer.scss]: why global?

now all h1 and p have 0 margin.

Do it only for specific place

**Comment by ShevchukTania** [comment]
проглянуто

**Review by igorkasyanchuk — APPROVED** [review]

**Pull request merged**